File: arrhash/kfreq.py
nums = [1,2,2,2,3,3,3,4,5,6,6,9]
k = 2

def solution(nums, k):
  freq = {}
  for n in nums:
    freq[n] = freq.get(n, 0) + 1
  bucket = [[] for _ in range(len(nums) + 1)]
  for num, count in freq.items():
    bucket[count].append(num)

  res = []
  for i in range(len(bucket) - 1, 0, -1):
    for number in bucket[i]:
      res.append(number)
      if len(res) == k:
        return res

result = solution(nums, k)
print(result)

File: arrhash/kfreq.js
nums = [1, 2, 2, 2, 3, 3, 3, 4, 5, 6, 6, 9];
k = 2;

function topKFrequent(nums, k) {
  let freq = {};
  for (let n of nums) {
    freq[n] = (freq[n] || 0) + 1;
  }

  const bucket = Array.from({ length: nums.length + 1 }, () => []);
  for (const [number, count] of Object.entries(freq)) {
    bucket[count].push(Number(number));
  }

  let res = [];
  for (let i = bucket.length - 1; i > 0; i--) {
    for (let number of bucket[i]) {
      res.push(number);
      if (res.length === k) {
        return res;
      }
    }
  }
}

const result = topKFrequent(nums, k);
console.log(result);

document.getElementById("result").textContent = "Resultado: " + result;
